client/clientutil/modelinfo.go

```diff
@@ -0,0 +1,461 @@
+
+		// long string key we don't want to rewrap but can safely handle
+		// on "reasonable" width terminals
+		case "device-key-sha3-384":
+			// also flush the writer before continuing so the previous keys
+			// don't try to align with this key
+			w.Flush()
+			headerString, ok := headerValue.(string)
+			if !ok {
+				return invalidTypeErr
+			}
+
+			switch {
+			case options.TermWidth > 86:
+				fmt.Fprintf(w, "device-key-sha3-384: %s\n", headerString)
+			case options.TermWidth <= 86 && options.TermWidth > 66:
+				fmt.Fprintln(w, "device-key-sha3-384: |")
+				strutil.WordWrapPadded(w, []rune(headerString), "  ", options.TermWidth)
+			}
+		case "snaps":
+			// also flush the writer before continuing so the previous keys
+			// don't try to align with this key
+			w.Flush()
+			snapsHeader, ok := headerValue.([]interface{})
+			if !ok {
+				return invalidTypeErr
+			}
+			if len(snapsHeader) == 0 {
+				// unexpected why this is an empty list, but just ignore for
+				// now
+				continue
+			}
+			fmt.Fprintf(w, "snaps:\n")
+			for _, sn := range snapsHeader {
+				snMap, ok := sn.(map[string]interface{})
+				if !ok {
+					return invalidTypeErr
+				}
+				// iterate over all keys in the map in a stable, visually
+				// appealing ordering
+				// first do snap name, which will always be present since we
+				// parsed a valid assertion
+				name := snMap["name"].(string)
+				fmt.Fprintf(w, "  - name:\t%s\n", name)
+
+				// the rest of these may be absent, but they are all still
+				// simple strings
+				for _, snKey := range []string{"id", "type", "default-channel", "presence"} {
+					snValue, ok := snMap[snKey]
+					if !ok {
+						continue
+					}
+					snStrValue, ok := snValue.(string)
+					if !ok {
+						return invalidTypeErr
+					}
+					if snStrValue != "" {
+						fmt.Fprintf(w, "    %s:\t%s\n", snKey, snStrValue)
+					}
+				}
+
+				// finally handle "modes" which is a list
+				modes, ok := snMap["modes"]
+				if !ok {
+					continue
+				}
+				modesSlice, ok := modes.([]interface{})
+				if !ok {
+					return invalidTypeErr
+				}
+				if len(modesSlice) == 0 {
+					continue
+				}
+
+				modeStrSlice := make([]string, 0, len(modesSlice))
+				for _, mode := range modesSlice {
+					modeStr, ok := mode.(string)
+					if !ok {
+						return invalidTypeErr
+					}
+					modeStrSlice = append(modeStrSlice, modeStr)
+				}
+				modesSliceYamlStr := "[" + strings.Join(modeStrSlice, ", ") + "]"
+				fmt.Fprintf(w, "    modes:\t%s\n", modesSliceYamlStr)
+			}
+
+		// long base64 key we can rewrap safely
+		case "device-key":
+			headerString, ok := headerValue.(string)
+			if !ok {
+				return invalidTypeErr
+			}
+			// the string value here has newlines inserted as part of the
+			// raw assertion, but base64 doesn't care about whitespace, so
+			// it's safe to split by newlines and re-wrap to make it
+			// prettier
+			headerString = strings.Join(
+				strings.Split(headerString, "\n"),
+				"")
```

Comment:
```golang
headerString = strings.ReplaceAll(headerString, "\n", "")
```
seems simpler; or is the split + join combination more performant? (if so, I'd add a comment about it, otherwise someone else looking at the code would likely refactor it and use ReplaceAll)